presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -465,6 +485,23 @@ void fail(Throwable e)
                     splitBytes = internalSplit.getEnd() - internalSplit.getStart();
                 }
 
+                CacheQuota cacheQuota;
+                switch (cacheQuotaScope) {
+                    case GLOBAL:
+                        cacheQuota = new CacheQuota(".", configuredCacheQuota);
+                        break;
+                    case SCHEMA:
+                        cacheQuota = new CacheQuota(databaseName, configuredCacheQuota);
+                        break;
+                    case TABLE:
+                        cacheQuota = new CacheQuota(databaseName + "." + tableName, configuredCacheQuota);
+                        break;
+                    case PARTITION:
+                        cacheQuota = new CacheQuota(databaseName + "." + tableName + "." + internalSplit.getPartitionName(), configuredCacheQuota);
+                        break;
+                    default:
+                        throw new PrestoException(CACHE_SCOPE_NOT_SUPPORTED, format("%s is not supported", cacheQuotaScope));
```

Comment:
Use `HIVE_UNKNOWN_ERROR`

Reply:
I think it's more related to cache config not being supported, `HIVE_UNKNOWN_ERROR `  seems to be used more often for hive data related issues.

Also our next step might be breaking presto-cache and presto-hive, we need to add `CACHE_SCOPE_NOT_SUPPORTED ` back once that happens

Thus I think we can keep `CACHE_SCOPE_NOT_SUPPORTED ` but make it `EXTERNAL` like `HIVE_UNKNOWN_ERROR `

Reply:
We shouldn't create error codes for utilities. Error codes are meant for main engine or connectors. Check https://github.com/prestodb/presto/wiki/Error-Codes to see how we allocate error code space. If you really need to add a new error code, it should belong to HiveErrorCode